docs/proposals/managers-registry.md

```diff
@@ -0,0 +1,72 @@
+## Motivation
+
+In today implementation our _managers_ tends to have too much dependencies on each other, making them hard to tests.
+
+The example case is, adding a new feature in one manager, we have to refactor the tests of all the classes using it, same for registries.
+
+Moreover, we are initializing all the managers in the extension activate, which can be heavy, and decrease the performance of the Podman-Desktop application.
+
+## Solution
+
+We should introduce a `ManagerRegistry`. This class should be responsible for the lifecycle of the managers.
+
+![manager-registry-uml.png](manager-registry-uml.png)
+
+Inside the StudioAPI we can have an instance of the `ManagerRegistry` and calling the `getInstance` for each manager we require when the api is triggered.
+
+````ts
+import type { ModelsManager } from './modelsManager';
+
+... 
+
+async getModelsDirectory(): Promise <string> {
+  return this.managers.getInstance(ModelsManager).getModelsDirectory();
+}
```

Comment:
I would prefer to use this instead:
```
init() {
  this.modelsManager = this.managers.getInstance(ModelsManager);
  [...]
}

async getModelsDirectory(): Promise <string> {
  return this.modelsManager.getModelsDirectory();
}
```

This way, we won't need to rewrite all the calls to methods, but only the init methods. We can develop later if the init methods of the dependencies are too much heavy.

Reply:
I would agree at first to simply the usage maybe. But this remove the lazy loading we would get from this proposal

Reply:
I also think it would simplify the StudioApiImpl class to not keep those many private properties